src/components/Header/Header.tsx

```diff
@@ -0,0 +1,109 @@
+import classNames from 'classnames';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import { TodosContext } from '../TododsContext/TodosContext';
+import { addTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  setTempTodo: (todo: Todo | null) => void,
+};
+
+export const Header: React.FC<Props> = ({ setTempTodo }) => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const {
+    todos, setTodos, setErrorMessage, USER_ID, loading,
+  } = useContext(TodosContext);
+  const titleRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    titleRef.current?.focus();
+  }, [loading]);
+
+  const handleChangeTodoTitle = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setTodoTitle(event.target.value);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+  };
+
+  const handleChangeToggle = () => {
+    const completedValue = todos.some(todo => !todo.completed);
+
+    const updatedTodos = todos.map(todo => {
+      return {
+        ...todo,
+        completed: completedValue,
+      };
+    });
+
+    setTodos(updatedTodos);
+  };
+
+  const handlePressEnter = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      setTempTodo({
+        title: todoTitle,
+        completed: false,
+        userId: USER_ID,
+        id: 0,
+      });
+
+      if (todoTitle.trim() === '') {
+        setErrorMessage('Title should not be empty');
+        setTempTodo(null);
+      } else {
+        addTodo({
+          title: todoTitle,
+          completed: false,
+          userId: USER_ID,
+        })
+          .then((newTodo: Todo) => {
+            setTodos(currentTodos => [...currentTodos, newTodo]);
+            // titleRef.current?.focus();
```

Comment:
There's a commented-out line of code. If this line is not needed, consider removing it to keep the code clean.